File: gst-app/src/lib/indian-states.ts
// Indian state codes for GSTIN validation and place-of-supply determination

export interface IndianState {
    code: string;
    name: string;
    ut: boolean; // Union Territory
}

export const INDIAN_STATES: IndianState[] = [
    { code: '01', name: 'Jammu & Kashmir', ut: true },
    { code: '02', name: 'Himachal Pradesh', ut: false },
    { code: '03', name: 'Punjab', ut: false },
    { code: '04', name: 'Chandigarh', ut: true },
    { code: '05', name: 'Uttarakhand', ut: false },
    { code: '06', name: 'Haryana', ut: false },
    { code: '07', name: 'Delhi', ut: true },
    { code: '08', name: 'Rajasthan', ut: false },
    { code: '09', name: 'Uttar Pradesh', ut: false },
    { code: '10', name: 'Bihar', ut: false },
    { code: '11', name: 'Sikkim', ut: false },
    { code: '12', name: 'Arunachal Pradesh', ut: false },
    { code: '13', name: 'Nagaland', ut: false },
    { code: '14', name: 'Manipur', ut: false },
    { code: '15', name: 'Mizoram', ut: false },
    { code: '16', name: 'Tripura', ut: false },
    { code: '17', name: 'Meghalaya', ut: false },
    { code: '18', name: 'Assam', ut: false },
    { code: '19', name: 'West Bengal', ut: false },
    { code: '20', name: 'Jharkhand', ut: false },
    { code: '21', name: 'Odisha', ut: false },
    { code: '22', name: 'Chhattisgarh', ut: false },
    { code: '23', name: 'Madhya Pradesh', ut: false },
    { code: '24', name: 'Gujarat', ut: false },
    { code: '26', name: 'Dadra & Nagar Haveli and Daman & Diu', ut: true },
    { code: '27', name: 'Maharashtra', ut: false },
    { code: '28', name: 'Andhra Pradesh (old)', ut: false },
    { code: '29', name: 'Karnataka', ut: false },
    { code: '30', name: 'Goa', ut: false },
    { code: '31', name: 'Lakshadweep', ut: true },
    { code: '32', name: 'Kerala', ut: false },
    { code: '33', name: 'Tamil Nadu', ut: false },
    { code: '34', name: 'Puducherry', ut: true },
    { code: '35', name: 'Andaman & Nicobar Islands', ut: true },
    { code: '36', name: 'Telangana', ut: false },
    { code: '37', name: 'Andhra Pradesh', ut: false },
    { code: '38', name: 'Ladakh', ut: true },
    { code: '97', name: 'Other Territory', ut: false },
];

export const STATE_MAP = new Map(INDIAN_STATES.map(s => [s.code, s]));

export function getStateName(code: string): string | undefined {
    return STATE_MAP.get(code)?.name;
}

export function isValidStateCode(code: string): boolean {
    return STATE_MAP.has(code);
}

File: backend/app/services/indian_states.py
"""
Indian state codes for GSTIN validation and place-of-supply determination.
"""
from typing import Dict, Optional, List

INDIAN_STATES: List[Dict] = [
    {"code": "01", "name": "Jammu & Kashmir", "ut": True},
    {"code": "02", "name": "Himachal Pradesh", "ut": False},
    {"code": "03", "name": "Punjab", "ut": False},
    {"code": "04", "name": "Chandigarh", "ut": True},
    {"code": "05", "name": "Uttarakhand", "ut": False},
    {"code": "06", "name": "Haryana", "ut": False},
    {"code": "07", "name": "Delhi", "ut": True},
    {"code": "08", "name": "Rajasthan", "ut": False},
    {"code": "09", "name": "Uttar Pradesh", "ut": False},
    {"code": "10", "name": "Bihar", "ut": False},
    {"code": "11", "name": "Sikkim", "ut": False},
    {"code": "12", "name": "Arunachal Pradesh", "ut": False},
    {"code": "13", "name": "Nagaland", "ut": False},
    {"code": "14", "name": "Manipur", "ut": False},
    {"code": "15", "name": "Mizoram", "ut": False},
    {"code": "16", "name": "Tripura", "ut": False},
    {"code": "17", "name": "Meghalaya", "ut": False},
    {"code": "18", "name": "Assam", "ut": False},
    {"code": "19", "name": "West Bengal", "ut": False},
    {"code": "20", "name": "Jharkhand", "ut": False},
    {"code": "21", "name": "Odisha", "ut": False},
    {"code": "22", "name": "Chhattisgarh", "ut": False},
    {"code": "23", "name": "Madhya Pradesh", "ut": False},
    {"code": "24", "name": "Gujarat", "ut": False},
    {"code": "26", "name": "Dadra & Nagar Haveli and Daman & Diu", "ut": True},
    {"code": "27", "name": "Maharashtra", "ut": False},
    {"code": "28", "name": "Andhra Pradesh (old)", "ut": False},
    {"code": "29", "name": "Karnataka", "ut": False},
    {"code": "30", "name": "Goa", "ut": False},
    {"code": "31", "name": "Lakshadweep", "ut": True},
    {"code": "32", "name": "Kerala", "ut": False},
    {"code": "33", "name": "Tamil Nadu", "ut": False},
    {"code": "34", "name": "Puducherry", "ut": True},
    {"code": "35", "name": "Andaman & Nicobar Islands", "ut": True},
    {"code": "36", "name": "Telangana", "ut": False},
    {"code": "37", "name": "Andhra Pradesh", "ut": False},
    {"code": "38", "name": "Ladakh", "ut": True},
    {"code": "97", "name": "Other Territory", "ut": False},
]

STATE_MAP: Dict[str, Dict] = {s["code"]: s for s in INDIAN_STATES}


def get_state_name(code: str) -> Optional[str]:
    state = STATE_MAP.get(code)
    return state["name"] if state else None


def is_valid_state_code(code: str) -> bool:
    return code in STATE_MAP
